clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -148,6 +158,32 @@ class AnnotatingParser {
     }
   }
 
+  const FormatStyle::FunctionDeclarationWithKeywords *
+  findFunctionWithKeywordedParameters() const {
+    const FormatToken *TokBeforeFirstLParent{};
+    for (const FormatToken *T = Line.First; T != Line.Last; T = T->Next) {
+      if (T->TokenText == StringRef("(", 1)) {
```

Comment:
does it need to be found by string? could it be by T->is(l_paren)

Reply:
At the point this function is called the paren token is still of type `tok::unknown`,  while I don't know  a suitable place to move the call to `findFunctionWithKeywordedParameters()` to.

Reply:
It's been a while, since I last stepped through the code myself. But are you really sure about that? It's not `clang-format` (or `libFormat`) that assigns the token type.